### ISSUE TITLE
RustDesk

### ISSUE THREAD

**Issue by ALipAkrohk** [chaotic-aur/packages#928]
## 👴 For reporting outdated packages

[rustdesk on aur](https://aur.archlinux.org/packages/rustdesk/) : [RustDesk.log on chaotic-aur](https://builds.garudalinux.org/repos/chaotic-aur/logs/rustdesk.log)
[Github Releases List](https://github.com/rustdesk/rustdesk/releases)

- Package name: `RustDesk`

- Latest build: `v1.1.6-1`

- Latest version available: `v1.1.8-1`

- Have you tested if the latest version of the package builds in a clean chroot?:

  - [x] YES
  - [ ] NO

## 🐛 For reporting bugs

- What happens?

 After installing the version available in the chaotic-aur repository, the program practically does not run.

- What is expected to happen?

Graphical interface of the software to run and remote desktop capability to be active.

**Comment by Technetium1** [comment]
```bash
$ rustdesk
thread 'main' panicked at 'error: 'libsciter-gtk.so' was not found neither in PATH nor near the current executable.
Please verify that Sciter SDK is installed and its binaries (from bin.lnx) are available in PATH.', /home/main-builder/.cargo/git/checkouts/rust-sciter-06aa50f9c0fcf3d6/82025b9/src/lib.rs:359:21
```

**Comment by Technetium1** [comment]
My friend has taken ownership of the `rustdesk` [package](https://aur.archlinux.org/packages/rustdesk) and will hopefully be able to fix the issue with `rustdeskt-git` in the process.

**Comment by dr460nf1r3** [comment]
Finally, it happened! 🥳

**Comment by Technetium1** [comment]
It's a really stupid fix too, no thanks to the maintainer either, they call *ONLY* building on the 2020 version a non-issue! At least it works for now:tm:

**Comment by joe733** [comment]
This has cropped up again. Not sure if  [`libsciter-gtk-git-bin`](https://aur.archlinux.org/packages/libsciter-gtk-git-bin) is needed.

```rust
~ 
➜ rustdesk 
thread 'main' panicked at 'error: 'libsciter-gtk.so' was not found neither in PATH nor near the current executable.
  Please verify that Sciter SDK is installed and its binaries (from bin.lnx) are available in PATH.', /home/main-builder/.cargo/git/checkouts/rust-sciter-06aa50f9c0fcf3d6/82025b9/src/lib.rs:359:21
stack backtrace:
[2022-08-05T07:49:08Z INFO  rustdesk::server] DISPLAY=Ok(":0.0")
[2022-08-05T07:49:08Z INFO  rustdesk::server] XAUTHORITY=Ok("~/.Xauthority")
[2022-08-05T07:49:08Z INFO  rustdesk::server] server not started (will try to start): No such file or directory (os error 2)
[2022-08-05T07:49:08Z INFO  rustdesk::server] DISPLAY=Ok(":0.0")
[2022-08-05T07:49:08Z INFO  rustdesk::server] XAUTHORITY=Ok("~/.Xauthority")
[2022-08-05T07:49:08Z INFO  rustdesk::ipc] Started ipc server at path: /tmp/RustDesk/ipc
[2022-08-05T07:49:08Z INFO  rustdesk::common] Testing nat ...
[2022-08-05T07:49:08Z INFO  rustdesk::rendezvous_mediator] lan discovery listener started
[2022-08-05T07:49:08Z INFO  rustdesk::rendezvous_mediator] start rendezvous mediator of rs-ny.rustdesk.com
[2022-08-05T07:49:08Z INFO  rustdesk::rendezvous_mediator] start rendezvous mediator of rs-sg.rustdesk.com
[2022-08-05T07:49:08Z INFO  rustdesk::rendezvous_mediator] start rendezvous mediator of rs-cn.rustdesk.com
   0: rust_begin_unwind
             at /rustc/1.62.1/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/1.62.1/library/core/src/panicking.rs:142:14
   2: core::panicking::panic_display
   3: std::sync::once::Once::call_once::{{closure}}
   4: std::sync::once::Once::call_inner
             at /rustc/1.62.1/library/std/src/sync/once.rs:434:21
   5: sciter::set_options
   6: rustdesk::ui::start
   7: rustdesk::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
``

**Comment by Technetium1** [comment]
We'll get that fixed ASAP
cc @Rojikku

**Comment by Rojikku** [comment]
See. I know that's an issue. But I don't know why that's an issue. It popped up when I modified the pkgbuild to fix the desktop file - or at least, I realized it when compiling then. Problem is the file is very obviously included in the program directory, so I guess the question is why it isn't looking for it.
I could mark it as a dependency, which would work once installed, but it previously wasn't building with that dependency, I had to do it manually. And doing both seems kind of silly.

I'll look into it though, but I'm inclined to believe the application is just weird.

**Comment by Rojikku** [comment]
TL;DR: Fixed++

So the issue...I don't know, but I know I can fix it by adding the path the .so file is in to /etc/ld.so.conf.d in a new config file.
Problem is there's apparently also a white screen issue. I don't know what idiot thought loading the UI from `$(pwd)/src/ui/index.html` was a good idea. It's very easy to inject your own UI into the program doing this.
I see rustdesk-git resolved this issue by moving everything into /usr/shared/rustdesk and then running from there. I roughly emulated that behavior.

As I was doing that, and current directory is usually in the path, I removed my ld.so.conf override and put libsciter into the /usr/shared as well, which rustdesk-git also did. As such, you will still experience this issue if you actually run the executable instead of the bash script.

Also, yes, using `libsciter-gtk-git-bin` would be a much better solution that loading it myself. However, that does not appear to build, and I am not entirely sure why. It complained about pkgver but I can't see the issue with my eyes.

Anyway, should work fine now.

**Comment by dr460nf1r3** [comment]
Let's see, I just started a build - fingers crossed! 😁

**Comment by Technetium1** [comment]
Just [deployed](https://t.me/chaotic_aur_deploy_log/238429) newest version and confirmed fix :heavy_check_mark:

**Comment by tio-trom** [comment]
Was https://aur.archlinux.org/packages/rustdesk removed? Can't find it here https://builds.garudalinux.org/repos/chaotic-aur/logs/filtered/build-failed/ or here https://builds.garudalinux.org/repos/chaotic-aur/logs/filtered/dep-not-in-repo/

**Comment by dr460nf1r3** [comment]
https://github.com/chaotic-aur/packages/issues/2086

**Comment by dr460nf1r3** [comment]
Once again, there seem to be compilation issues with this package:
```
   Compiling zstd v0.9.2+zstd.1.5.1
error: non-binding let on a synchronization lock
   --> libs/hbb_common/src/config.rs:659:13
    |
659 |         let _ = CONFIG.read().unwrap(); // for lock
    |             ^   ^^^^^^^^^^^^^^^^^^^^^^ this binding will immediately drop the value assigned to it
    |             |
    |             this lock is not assigned to a binding and is immediately dropped
    |
    = note: `#[deny(let_underscore_lock)]` on by default
help: consider binding to an unused variable to avoid immediately dropping the value
    |
659 |         let _unused = CONFIG.read().unwrap(); // for lock
    |             ~~~~~~~
help: consider immediately dropping the value
    |
659 |         drop(CONFIG.read().unwrap()); // for lock
    |         ~~~~~                      +

error: non-binding let on a synchronization lock
   --> libs/hbb_common/src/config.rs:670:13
    |
670 |         let _ = CONFIG.read().unwrap(); // for lock
    |             ^   ^^^^^^^^^^^^^^^^^^^^^^ this binding will immediately drop the value assigned to it
    |             |
    |             this lock is not assigned to a binding and is immediately dropped
    |
help: consider binding to an unused variable to avoid immediately dropping the value
    |
670 |         let _unused = CONFIG.read().unwrap(); // for lock
    |             ~~~~~~~
help: consider immediately dropping the value
    |
670 |         drop(CONFIG.read().unwrap()); // for lock
    |         ~~~~~                      +

error: non-binding let on a synchronization lock
   --> libs/hbb_common/src/config.rs:811:13
    |
811 |         let _ = CONFIG.read().unwrap(); // for lock
    |             ^   ^^^^^^^^^^^^^^^^^^^^^^ this binding will immediately drop the value assigned to it
    |             |
    |             this lock is not assigned to a binding and is immediately dropped
    |
help: consider binding to an unused variable to avoid immediately dropping the value
    |
811 |         let _unused = CONFIG.read().unwrap(); // for lock
    |             ~~~~~~~
help: consider immediately dropping the value
    |
811 |         drop(CONFIG.read().unwrap()); // for lock
    |         ~~~~~                      +

error: could not compile `hbb_common` due to 3 previous errors
warning: build failed, waiting for other jobs to finish...
error: could not compile `hbb_common` due to 3 previous errors
==> ERROR: A failure occurred in build().
```

**Comment by Technetium1** [comment]
Upstream issue filed: https://github.com/rustdesk/rustdesk/issues/2087

**Comment by Technetium1** [comment]
The dev was about as helpful as they have been historically…

**Comment by dr460nf1r3** [comment]
Oh yes. I'm considering dropping it for this exact reason - can't remember a longer period of time this hasn't caused issues.

**Comment by dr460nf1r3** [comment]
Oops, marked @Technetium1's issue number instead of this one.. https://github.com/chaotic-aur/packages/commit/b810999be5df443f84071dd3e64f22c0eeb1ce78

**Comment by tio-trom** [comment]
Guys how about https://aur.archlinux.org/packages/rustdesk-bin ?

**Comment by Technetium1** [comment]
@tio-trom maybe if it's updated for the new version. For now there is `rustdesk-nightly` from the above commit. That is the only one that's gotten required patches upstream (for now), since the maintainer doesn't enjoy creating tags other than when deadly necessary.

**Comment by tio-trom** [comment]
Isn't https://github.com/rustdesk/rustdesk/releases/tag/1.1.9 the latest stable?

**Comment by Rojikku** [comment]
> Isn't https://github.com/rustdesk/rustdesk/releases/tag/1.1.9 the latest stable?

Yes. That's the issue. The developer has the fix in nightly but hasn't released it officially.

**Comment by tio-trom** [comment]
Ah ok....

**Comment by xiota** [comment]
Dropping because unusual packaging and out of date.  `rustdesk-bin` is available.  Metrics:

* rustdesk-nightly 7
* rustdesk-appimage-nightly 15
